pontoon/sync/core/stats.py

```diff
@@ -0,0 +1,197 @@
+                        SUM(tr.total_strings) AS "total",
+                        SUM(tr.approved_strings) AS "approved",
+                        SUM(tr.pretranslated_strings) AS "pretranslated",
+                        SUM(tr.strings_with_errors) AS "errors",
+                        SUM(tr.strings_with_warnings) AS "warnings",
+                        SUM(tr.unreviewed_strings) AS "unreviewed"
+                    FROM "base_translatedresource" tr
+                    INNER JOIN "base_resource" res ON (tr.resource_id = res.id)
+                    WHERE res.project_id = %s
+                    GROUP BY tr.locale_id
+                ) AS agg
+                WHERE agg.locale_id = pl.locale_id AND pl.project_id = %s
+                """
+            ),
+            [project.id, project.id],
+        )
+        pl_count = cursor.rowcount
+
+        # Project, counted from project locales
+        cursor.execute(
+            dedent(
+                """
+                UPDATE base_project proj
+                SET
+                    total_strings = GREATEST(agg.total, 0),
+                    approved_strings = GREATEST(agg.approved, 0),
+                    pretranslated_strings = GREATEST(agg.pretranslated, 0),
+                    strings_with_errors = GREATEST(agg.errors, 0),
+                    strings_with_warnings = GREATEST(agg.warnings, 0),
+                    unreviewed_strings = GREATEST(agg.unreviewed, 0)
+                FROM (
+                    SELECT
+                        SUM(pl.total_strings) AS "total",
+                        SUM(pl.approved_strings) AS "approved",
+                        SUM(pl.pretranslated_strings) AS "pretranslated",
+                        SUM(pl.strings_with_errors) AS "errors",
+                        SUM(pl.strings_with_warnings) AS "warnings",
+                        SUM(pl.unreviewed_strings) AS "unreviewed"
+                    FROM "base_projectlocale" pl
+                    WHERE pl.project_id = %s
+                ) AS agg
+                WHERE proj.id = %s
+                """
+            ),
+            [project.id, project.id],
+        )
+
+        lc_count = _update_locales(cursor) if update_locales else 0
+
+    tr_str = (
+        "1 translated resource" if tr_count == 1 else f"{tr_count} translated resources"
+    )
+    pl_str = "1 projectlocale" if pl_count == 1 else f"{pl_count} projectlocales"
+    lc_str = "1 locale" if lc_count == 1 else f"{lc_count} locales"
+    summary = (
+        f"{tr_str} and {pl_str}"
+        if lc_count == 0
+        else f"{tr_str}, {pl_str}, and {lc_str}"
+    )
+    log.info(f"[{project.slug}] Updated stats for {summary}")
+
+
+def update_locale_stats() -> None:
+    with connection.cursor() as cursor:
+        lc_count = _update_locales(cursor)
+    lc_str = "1 locale" if lc_count == 1 else f"{lc_count} locales"
+    log.info(f"Updated stats for {lc_str}")
+
+
+def _update_locales(cursor) -> int:
```

Comment:
You could do this with Django ORM:
* Aggregate `ProjectLocale` data
* Update `Locale` instances using `bulk_update()`

Reply:
But wait... Doesn't really make much sense given `update_stats()`, which will be very hard to port over.